.github/workflows/ci.yaml

```diff
@@ -120,7 +120,10 @@ jobs:
       - uses: Swatinem/rust-cache@v2
       - name: "Run tests"
         shell: bash
-        run: cargo nextest run --workspace --status-level skip --failure-output immediate-final --no-fail-fast -j 12
```

Comment:
I removed the `-j 12`. It gives us minimal speedup but has the downside that it requires manual updating (that likely doesn't happen) when any properties about the runners change.

Reply:
Did you benchmark it? It seemed like it was about 30s faster for WIndows (https://github.com/astral-sh/ruff/pull/9921#issuecomment-1937094551).

Reply:
I did not but you mentioned on your PR that it is minimal :laughing:

Reply:
Lol you got me

Reply:
I ended up keeping it because the Windows difference was kinda big though...